### DISCUSSION TITLE
How to reuse connections between different accounts?

### DISCUSSION THREAD

**Discussion by Taowyoo** [awslabs/aws-sdk-rust#1067] [General]
# Use case
I want to scan multiple accounts (100+) in different regions, and I want to reuse TLS connections for each region, which means no matter how many accounts I want to access, the code only creates TLS connections per region.

# Problem
For example, I want to make a lot of requests for account A and B in 5 regions. I would expect there are only 5 TLS handshake happened but not. 
I already ensure only one [SharedHttpClient](https://docs.rs/aws-smithy-runtime-api/1.1.5/aws_smithy_runtime_api/client/http/struct.SharedHttpClient.html) is created and used in different [aws_config::SdkConfig](https://docs.rs/aws-config/latest/aws_config/struct.SdkConfig.html#)

# Question
How could I achieve it or is it possible?

**Comment by rcoh** [comment]
Interesting. I would expect that this would work. Can you send some more code detailing how you set things up? Just a single service right?

Can you also explain how you determined that multiple TLS connections were being established?

**Reply by Taowyoo** [reply]
Thank you for reply. Finally, after upgrade to use latest version of SDK, there is no too much TLS connections created in our usecase.
